src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SecurityChannelFactory.cs

```diff
@@ -0,0 +1,845 @@
+            }
+            else if (typeof(TChannel) == typeof(IRequestChannel))
+            {
+                return (TChannel)(object)new SecurityRequestChannel(this, _securityProtocolFactory, ((IChannelFactory<IRequestChannel>)InnerChannelFactory).CreateChannel(address, via), address, via);
+            }
+
+            //typeof(TChannel) == typeof(IRequestSessionChannel)
+            return (TChannel)(object)new SecurityRequestSessionChannel(this, _securityProtocolFactory, ((IChannelFactory<IRequestSessionChannel>)InnerChannelFactory).CreateChannel(address, via), address, via);
+        }
+
+        protected override void OnOpen(TimeSpan timeout)
+        {
+            TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+            OnOpenCore(timeoutHelper.RemainingTime());
+            base.OnOpen(timeoutHelper.RemainingTime());
+            SetBufferManager();
+        }
+
+        protected internal override Task OnOpenAsync(TimeSpan timeout)
+        {
+            return OnOpenAsyncInternal(timeout);
+        }
+
+        private async Task OnOpenAsyncInternal(TimeSpan timeout)
+        {
+            TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+            await OnOpenCoreAsync(timeoutHelper.RemainingTime());
+            await base.OnOpenAsync(timeoutHelper.RemainingTime());
+            SetBufferManager();
+        }
+
+        void SetBufferManager()
+        {
+            ITransportFactorySettings transportSettings = GetProperty<ITransportFactorySettings>();
+
+            if (transportSettings == null)
+                return;
+
+            BufferManager bufferManager = transportSettings.BufferManager;
+
+            if (bufferManager == null)
+                return;
+
+            if (SessionMode && SessionClientSettings != null && SessionClientSettings.SessionProtocolFactory != null)
+            {
+                SessionClientSettings.SessionProtocolFactory.StreamBufferManager = bufferManager;
+            }
+            else
+            {
+                ThrowIfProtocolFactoryNotSet();
+                _securityProtocolFactory.StreamBufferManager = bufferManager;
+            }
+        }
+
+
+        protected override IAsyncResult OnBeginOpen(TimeSpan timeout, AsyncCallback callback, object state)
+        {
+            throw ExceptionHelper.PlatformNotSupported(); // Issue #31 in progress
+            //return new OperationWithTimeoutAsyncResult(new OperationWithTimeoutCallback(OnOpen), timeout, callback, state);
+        }
+
+        protected override void OnEndOpen(IAsyncResult result)
+        {
+            throw ExceptionHelper.PlatformNotSupported(); // Issue #31 in progress
+            //OperationWithTimeoutAsyncResult.End(result);
+        }
+
+        void OnOpenCore(TimeSpan timeout)
```

Comment:
Agree. I opened issue #1528 to do the all-up CodeFormatter pass.

---
In reply to: [78679251](https://github.com/dotnet/wcf/pull/1515#discussion_r78679251) [](ancestors = 78679251)